onnxruntime/core/providers/dml/OperatorAuthorHelper/OperatorHelper.cpp

```diff
@@ -2480,6 +2481,50 @@ namespace OperatorHelper
                 MLOperatorTensor outputSizesTensor = kernelInformation.GetConstantInputTensor(3);
                 ReadCpuLocalTensorIntoInt32(outputSizesTensor, /*out*/ outputSizes);
             }
+
+            axes = kernelInformation.GetAttributes().GetOptionalAttributeVectorInt32(AttrName::Axes);
+            // Handle possible axes input
+            if (opsetVersion >= 18 && !axes.empty())
+            {
+                uint32_t dimCount = gsl::narrow_cast<uint32_t>(m_inputDimensions.size());
+                HandleEmptyAxes(axes, m_inputDimensions, false);
+                HandleNegativeAxes(axes, dimCount);
+
+                // Taken from https://github.com/onnx/onnx/blob/3d69db8fd16873d68e7033479467f9478562a12d/onnx/reference/ops/op_resize.py#L303
+                if (!m_scales.empty())
+                {
+                    std::vector<float> newScales(dimCount, 1.f);
+                    // Fill in roi and scales/sizes
+                    uint32_t numAxes = gsl::narrow_cast<uint32_t>(axes.size());
+                    for (int32_t i = 0; i < axes.size(); i++)
+                    {
+                        newScales.at(axes[i]) = m_scales[i];
+                    }
+                    m_scales = newScales;
+                }
+                if (!outputSizes.empty())
+                {
+                    std::vector<int32_t> newSizes(m_inputDimensions.size(), 0);
+                    std::transform(m_inputDimensions.begin(), m_inputDimensions.end(), newSizes.begin(), [](auto v) {return static_cast<int32_t>(v);});
```

Comment:
Rather than explicitly call std::transform, can we use existing helpers like `DowncastDimensions` or templatize it? Alternately consider changing outputSizes to uint32_t (since there can't be negative sizes anyway) and adding a variant of `ReadCpuLocalTensorIntoInt32` or templatizing the output parameter on it.